tests/hfmagpsu.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+from utils.channel_access import ChannelAccess
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+from utils.ioc_launcher import IOCRegister, get_default_ioc_dir
+import time
+
+DEVICE_PREFIX = "HFMAGPSU_01"
+
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("HFMAGPSU"),
+        "macros": {},
+        "emulator": "HFMAGPSU"
+    },
+]
+
+
+class EnumValues(object):
+    AMPS = 0
+    TESLA = 1
+    ON = 1
+    OFF = 0
+
+
+class StringValues(object):
+    TESLA = "TESLA"
+    AMPS = "AMPS"
+    ON = "ON"
+    OFF = "OFF"
+
+TEST_MODES = [TestModes.DEVSIM, TestModes.RECSIM]
+
+
+class HfmagpsuTests(unittest.TestCase):
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("HFMAGPSU", DEVICE_PREFIX)
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX, default_timeout=30)
+        self.ca.wait_for("DIRECTION")
+        time.sleep(8)
+
+    def test_GIVEN_midTarget_set_WHEN_read_THEN_midTarget_is_as_expected(self):
+        test_value = 2.325

```

Comment:
Are your this test value representative of values the PV will take? Does it work with exponential notation e.g. `2e-2`? 

You can use [parameterized](https://github.com/wolever/parameterized) to parameterize your tests. Just pip install parameterized. This is going to be added to genie python install script (see https://github.com/ISISComputingGroup/genie_python/pull/135).